lib/child_process.js

```diff
@@ -60,22 +60,29 @@ function stdioStringToArray(option) {
 }
 
 exports.fork = function fork(modulePath /* , args, options */) {
-
   // Get options and args arguments.
   var execArgv;
-  var options = {};
-  var args = [];
-  var pos = 1;
-  if (pos < arguments.length && Array.isArray(arguments[pos])) {
-    args = arguments[pos++];
-  }
 
-  if (pos < arguments.length && arguments[pos] != null) {
-    if (typeof arguments[pos] !== 'object') {
-      throw new ERR_INVALID_ARG_VALUE(`arguments[${pos}]`, arguments[pos]);
-    }
+  var args = arguments[1];
+  var options = arguments[2];
+
+  if (Array.isArray(args)) {
+    args = args.slice(0);
```

Comment:
Is the slice needed? Please add a test case for it and remove it if there is no need for it.

Reply:
I'm not sure. But for consistency I took inspiration from current implementation at -

https://github.com/nodejs/node/blob/468ab4519e1b92473acefb22801497a1af6aebae/lib/child_process.js#L403

Reply:
That is not a strong argument. Please just verify this by adding a test as suggested.